sdk/state/state_test.go

```diff
@@ -391,6 +414,177 @@ func Test(t *testing.T) {
 	assert.EqualValues(t, initiatorEscrowResponse.Balances[0].Balance, fmt.Sprintf("%.7f", float64(iBalanceCheck)/float64(1_000_0000)))
 }
 
+func TestOpenUpdatesCoordinatedClose(t *testing.T) {
+	initiator, responder := initAccounts(t)
+	initiatorChannel, responderChannel := initChannels(t, initiator, responder)
+
+	s := initiator.EscrowSequenceNumber + 1
+	i := int64(1)
+	e := int64(0)
+	t.Log("Vars: s:", s, "i:", i, "e:", e)
+
+	// Open
+	t.Log("Open...")
+	open, err := initiatorChannel.ProposeOpen()
+	require.NoError(t, err)
+	for {
+		var fullySignedR bool
+		open, fullySignedR, err = responderChannel.ConfirmOpen(open)
+		if err != nil {
+			t.Fatal(err)
+		}
+		var fullySignedI bool
+		open, fullySignedI, err = initiatorChannel.ConfirmOpen(open)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if fullySignedI && fullySignedR {
+			break
+		}
+	}
+
+	{
+		ci, di, fi, err := initiatorChannel.OpenTxs()
+		require.NoError(t, err)
+
+		ci, err = ci.AddSignatureDecorated(open.CloseSignatures...)
+		require.NoError(t, err)
+
+		di, err = di.AddSignatureDecorated(open.DeclarationSignatures...)
+		require.NoError(t, err)
+
+		fi, err = fi.AddSignatureDecorated(open.FormationSignatures...)
+		require.NoError(t, err)
+
+		fbtx, err := txnbuild.NewFeeBumpTransaction(txnbuild.FeeBumpTransactionParams{
+			Inner:      fi,
+			FeeAccount: initiator.KP.Address(),
+			BaseFee:    txnbuild.MinBaseFee,
+		})
+		require.NoError(t, err)
+		fbtx, err = fbtx.Sign(networkPassphrase, initiator.KP)
+		require.NoError(t, err)
+		_, err = client.SubmitFeeBumpTransaction(fbtx)
+		require.NoError(t, err)
+	}
+
+	// Perform a number of iterations, much like two participants may.
+	// Exchange signed C_i and D_i for each
+	t.Log("Subsequent agreements...")
+	rBalanceCheck := responder.Contribution
+	iBalanceCheck := initiator.Contribution
+	for i < 20 {
```

Comment:
@acharb this seems like non-deterministic testing, which doesn't make it very easy to replicate if it every fails.

Is there a way to make this into a deterministic testing pattern?
If yes, then can we add a TODO to convert this to a deterministic testing pattern?

tagging @leighmcculloch to get your thoughts as well.

Reply:
i don't feel strongly here. The only non-deterministic parts of these integration tests are the payment amounts and that we're calling a local standalone network. I think for this use case and stage of the project that makes sense, but I'd be happy to discuss further. 

We have a ticket [here](https://github.com/stellar/experimental-payment-channels/issues/70) for adding unit testing which won't rely on a network and can have non-random payment amounts to be deterministic